### DISCUSSION TITLE
Extension operation disappears for some queries in rdf4j v4

### DISCUSSION THREAD

**Discussion by apano-on** [eclipse-rdf4j/rdf4j#3991] [Q&A] [answered]
It seems like the operation with the extension has disappeared in at least one scenario for rdf4j v4.0.0 (as well as v4.0.2). Is there a rationale for this change?

### Steps to reproduce
The following query is translated differently in the new version of rdf4j
```
SELECT (?l AS ?v)
WHERE { ?s rdfs:label ?l . }
ORDER BY ?v
```

### Current Behavior (rdf4j v4.0.2 translation/algebra tree)
```
QueryRoot
   Projection
      ProjectionElemList
         ProjectionElem "l" AS "v"
      Order
         OrderElem (ASC)
            Var (name=v)
         StatementPattern
            Var (name=s)
            Var (name=_const_9285ccfc_uri, value=http://www.w3.org/2000/01/rdf-schema#label, anonymous)
            Var (name=l)
```

### Expected Behavior (rdf4j v3 translation/algebra tree)
```
Projection
   ProjectionElemList
      ProjectionElem "l" AS "v"
   Order
      OrderElem (ASC)
         Var (name=v)
      Extension
         ExtensionElem (v)
            Var (name=l)
         StatementPattern
            Var (name=s)
            Var (name=_const_9285ccfc_uri, value=http://www.w3.org/2000/01/rdf-schema#label, anonymous)
            Var (name=l)
```

**Comment by hmottestad** [comment, marked as answer]
See https://github.com/eclipse/rdf4j/issues/2990 and https://github.com/eclipse/rdf4j/pull/3705

**Reply by apano-on** [reply]
For the example provided in the referenced issue, the simplification / redundant extension node works.

However, what about queries like the one I added in my comment above which contain an Order By. When the Order By is defined on a variable which is aliased (e.g. v above), without the Extension that variable is no longer in the scope of the respective subtree. It was the Extension node/operator which tracked the aliasing.

Doesn't this simplification break the tree when there is aliasing AND an Order By?

**Reply by hmottestad** [reply]
@jeenbroekstra do you have time to time to look into this?

**Reply by abrokenjester** [reply]
I'm afraid I have no availability until late next week. But happy to take a closer look after that.

**Reply by abrokenjester** [reply]
Logged bug #4066
